pkg/server/image_pull.go

```diff
@@ -81,7 +82,7 @@ import (
 // contents are missing but snapshots are ready, is the image still "READY"?
 
 // PullImage pulls an image with authentication config.
-func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest) (*runtime.PullImageResponse, error) {
+func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest) (_ *runtime.PullImageResponse, err error) {
```

Comment:
IIRC the `getResolver` function below also does some network requests (not very sure about that though). If that's the case then can we put that into the conditional block too?

Reply:
Into the `DisableSameImageParallelPull` conditional?

Reply:
I'm not following really

Reply:
As in let only one goroutine do that. Others can simply wait.

Reply:
Ahh, I'm not too worried about optimizing anything before the main pull work. So long as the blob downloads get skipped (part of pull) and the unpacking, I think that's more than enough for this change, but let me take a closer look at what `getResolver` does under the hood.

Reply:
I looked into this and yes it makes a request to get the manifest/index. In Containerd upstream they changed this to just create a new docker Resolver object in this function so they can pass it to `containerd.WithResolver`. Maybe we should track down the commit that changed this and cherry-pick, but that can be in another PR.